src/init.cpp

```diff
@@ -1683,14 +1683,17 @@ bool AppInitMain(NodeContext& node, interfaces::BlockAndHeaderTipInfo* tip_info)
     }
 
     chainman.m_load_block = std::thread(&util::TraceThread, "loadblk", [=, &chainman, &args] {
-        ThreadImport(chainman, vImportFiles, ShouldPersistMempool(args) ? MempoolPath(args) : fs::path{});
+        // Import blocks
+        ImportBlocks(chainman, vImportFiles);
+        // Load mempool from disk
+        chainman.ActiveChainstate().LoadMempool(ShouldPersistMempool(args) ? MempoolPath(args) : fs::path{});
```

Comment:
> I'm confused about this commit because the commit message says it only "will allow us to start indexes prior loading the mempool" in future commits, but the actual code change in commit sets g_indexes_ready_to_sync before calling LoadMempool, so it seems like that change is happening here, not in future commits.

Upps, yeah. Sorry for that. I updated that commit description in a rush..

> Also renaming ImportBlocks to ThreadImport here makes it hard to see where the real changes are in this commit. I agree ImportBlocks would be a slightly better name than ThreadImport, but I also think ThreadImport is fine. What's more misleading than the function name is the thread name "loadblk", so if you do want to do a renaming, I'd suggest renaming both loadblk and ThreadImport in a separate scripted diff commit not mixed with more substantive changes.

Yeah ok. Agree.
Have renamed "loadblk" to "post_init_load" but I'm all ears if you have a better term for a thread that imports/load blocks, the mempool and starts indexes. No better naming has come to my head yet.

Reply:
> Yeah ok. Agree. Have renamed "loadblk" to "post_init_load" but I'm all ears if you have a better term for a thread that imports/load blocks, the mempool and starts indexes. No better naming has come to my head yet.

I think any name is besides current name `loadblk` is probably better just because the current name is misleading and the thread is doing more than loading block data. I'd probably call it `loadinit` just to suggest it is loading data and is related to initialization, so should not be a persistent thread. I think it's also good if the thread has a short name so it is readable in tools like `top`

Reply:
Ok done, renamed the thread to `initload`.